test/ch/faults_test.exs

```diff
@@ -168,16 +226,29 @@ defmodule Ch.FaultsTest do
       log =
         capture_async_log(fn ->
           {:ok, conn} = Ch.start_link(port: port)
+
+          # connect
           {:ok, mint} = :gen_tcp.accept(listen)
+
+          # handshake
+          :ok = :gen_tcp.send(clickhouse, intercept_packets(mint))
+          :ok = :gen_tcp.send(mint, intercept_packets(clickhouse))
+
+          # disconnect before insert
           :ok = :gen_tcp.close(mint)
 
           spawn_link(fn ->
             assert {:error, %Mint.TransportError{reason: :closed}} =
                      Ch.query(conn, "insert into example(a,b) format RowBinary", stream)
```

Comment:
I've received `{:error, %Mint.HTTP1{}}` here once. Didn't get the test seed. Can't reproduce it anymore...